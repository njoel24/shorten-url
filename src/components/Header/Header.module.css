.Header {
    display: flex;
    width: 100%;
}