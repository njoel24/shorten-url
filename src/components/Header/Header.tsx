import React from 'react';
import styles from './Header.module.css';
import TopHeader from './TopHeader/TopHeader';

const Header = () => (
  <header className={styles.Header} data-testid="Header">
    <TopHeader />
  </header>
);

export default Header;
