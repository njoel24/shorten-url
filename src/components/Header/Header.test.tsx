import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from './Header';

describe('<Header />', () => {
  test('it should mount', () => {
    render(<Header />);
    
    const header = screen.getByTestId('Header');

    expect(header).toBeInTheDocument();
  });

  test('it should match the snapshot', () => {
    const wrapper = render(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});