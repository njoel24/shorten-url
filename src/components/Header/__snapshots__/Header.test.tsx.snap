// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="Header"
        data-testid="Header"
      >
        <div
          class="HeaderWrapper"
        >
          <div
            class="TopHeader"
            data-testid="TopHeader"
          >
            <div
              class="Left"
            >
              <svg>
                logo.svg
              </svg>
              <ul
                class="TopHeaderMenu"
                data-testid="Menu"
              >
                <li>
                  Features
                </li>
                <li>
                  Pricing
                </li>
                <li>
                  Resources
                </li>
              </ul>
            </div>
            <ul
              class="Account"
              data-testid="Account"
            >
              <button
                class="Button"
              >
                Login
              </button>
              <button
                class="Button active"
              >
                Sign Up
              </button>
            </ul>
            <div
              class="AccountMenuMobile"
              data-testid="AccountMenuMobile"
            >
              <div
                class="Separator"
              />
              <div
                class="AccountMenuMobileContent"
                id="AccountMenuMobile"
              >
                <ul
                  class="TopHeaderMenu"
                  data-testid="Menu"
                >
                  <li>
                    Features
                  </li>
                  <li>
                    Pricing
                  </li>
                  <li>
                    Resources
                  </li>
                </ul>
                <ul
                  class="Account"
                  data-testid="Account"
                >
                  <button
                    class="Button"
                  >
                    Login
                  </button>
                  <button
                    class="Button active"
                  >
                    Sign Up
                  </button>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="BottomHeader"
            data-testid="BottomHeader"
          >
            <div
              class="Right"
            >
              <svg>
                illustration-working.svg
              </svg>
            </div>
            <div
              class="Left"
            >
              <h1>
                More than just shorten links
              </h1>
              <h6>
                Build your brand's recognition and get detailed insight on how your links are performing
              </h6>
              <button
                class="Button large active"
                data-testid="BottomHeaderButton"
              >
                Get Started
              </button>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="Header"
      data-testid="Header"
    >
      <div
        class="HeaderWrapper"
      >
        <div
          class="TopHeader"
          data-testid="TopHeader"
        >
          <div
            class="Left"
          >
            <svg>
              logo.svg
            </svg>
            <ul
              class="TopHeaderMenu"
              data-testid="Menu"
            >
              <li>
                Features
              </li>
              <li>
                Pricing
              </li>
              <li>
                Resources
              </li>
            </ul>
          </div>
          <ul
            class="Account"
            data-testid="Account"
          >
            <button
              class="Button"
            >
              Login
            </button>
            <button
              class="Button active"
            >
              Sign Up
            </button>
          </ul>
          <div
            class="AccountMenuMobile"
            data-testid="AccountMenuMobile"
          >
            <div
              class="Separator"
            />
            <div
              class="AccountMenuMobileContent"
              id="AccountMenuMobile"
            >
              <ul
                class="TopHeaderMenu"
                data-testid="Menu"
              >
                <li>
                  Features
                </li>
                <li>
                  Pricing
                </li>
                <li>
                  Resources
                </li>
              </ul>
              <ul
                class="Account"
                data-testid="Account"
              >
                <button
                  class="Button"
                >
                  Login
                </button>
                <button
                  class="Button active"
                >
                  Sign Up
                </button>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="BottomHeader"
          data-testid="BottomHeader"
        >
          <div
            class="Right"
          >
            <svg>
              illustration-working.svg
            </svg>
          </div>
          <div
            class="Left"
          >
            <h1>
              More than just shorten links
            </h1>
            <h6>
              Build your brand's recognition and get detailed insight on how your links are performing
            </h6>
            <button
              class="Button large active"
              data-testid="BottomHeaderButton"
            >
              Get Started
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
